syntax/yang.vim

```diff
@@ -6,92 +6,94 @@
+syntax keyword yangStatement leaf leaf-list list notification prefix rpc typedef unique uses nextgroup=yangIdentifier skipwhite
 
 " other keywords
-syn keyword yangOther add current delete deprecated max min not-supported
-syn keyword yangOther obsolete replace system unbounded user
+syntax keyword yangOther add current delete deprecated max min not-supported
+syntax keyword yangOther obsolete replace system unbounded user
 
 " boolean constants (separated from the 'other keywords' for vim syntax purpose)
-syn keyword yangBoolean true false
+syntax keyword yangBoolean true false
 
 " if-feature (separated from 'statement keywords' for vim syntax purposes)
-syn keyword yangConditional if-feature
+syntax keyword yangConditional if-feature
 
 " comments
-syn region yangComment start=/\/\*/ end=/\*\//
-syn region yangComment start="//" end="$" 
+syntax region yangComment start="/\*" end="\*/"
+syntax region yangComment start="//" end="$"
 
 " strings
-syn region yangString start=+"+ skip=+\\\\\|\\"+ end=+"+
-syn region yangString start=+'+ skip=+\\\\\|\\'+ end=+'+
+syntax region yangString start=+"+ skip=+\\"+ end=+"+
+syntax region yangString start=+'+ end=+'+
 
 " dates
-syn match yangDateArg /"\=\<\d\{4}-\d\{2}-\d\{2}\>"\=/
+syntax match yangDateArg /"\=\<\d\{4}-\d\{2}-\d\{2}\>"\=/
 
 " length-arg TODO: this needs to also include the pipe and individual numbers (i.e. fixed length)
-syn match yangLengthArg /"\(\d\+\|min\)\s*\.\.\s*\(\d\+\|max\)"/
+syntax match yangLengthArg /"\(\d\+\|min\)\s*\.\.\s*\(\d\+\|max\)"/
 
 " numbers
-syn match yangNumber /\<[+-]\=\d\+\>/
-syn match yangNumber	"\<0x\x\+\>"
+syntax match yangNumber /\<[+-]\=\d\+\>/
+syntax match yangNumber	"\<0x\x\+\>"
 
 
 "-------------------------------------
 " and now for the highlighting
 
 " things with one-to-one mapping
-hi def link yangBadIdentifier Error
-hi def link yangIdentifier Identifier
-hi def link yangString String
-hi def link yangComment Comment
-hi def link yangNumber Number
-hi def link yangBoolean Boolean
-hi def link yangConditional Conditional
-hi def link yangType Type
+highlight def link yangBadIdentifier Error
+highlight def link yangIdentifier Identifier
+highlight def link yangString String
+highlight def link yangComment Comment
+highlight def link yangNumber Number
+highlight def link yangBoolean Boolean
+highlight def link yangConditional Conditional
+highlight def link yangType Type
+highlight def link yangKeyword Keyword
+highlight def link yangStatement Statement
 
 " arbitrary mappings
-hi def link yangKeyword Keyword
-hi def link yangStatement Statement
-hi def link yangModule Type
-hi def link yangDateArg Special
-hi def link yangLengthArg Special
+highlight def link yangModule Type
+highlight def link yangDateArg Special
+highlight def link yangLengthArg Special
 
-let b:current_syntax = "yang"
+" synchronize
+syntax sync match yangSync grouphere NONE '}$'
```

Comment:
I had noticed this problem where highlighting would get messed up if I jumped to the bottom of some yang files. Didn't know there was a solution. :tada:

Reply:
Yes, exactly - this is the one change that I care about, really!

In principle we can still be fooled by strings that contain lines that end in a `}`.  But in practice I've found that this is a big improvement, at least on the files that I've been looking at.